### PR TITLE
Add devcontainer support

### PR DESCRIPTION
**Pull request by SergioGasquez**
- Add devcontainer support for:
  - VS Code Dev Containers
  - Gitpod
  - GitHub Codespaces
- When using dev containers:
  - Wokwi support
  - Web flasher support
  - Readme with instructions

### CODE REVIEW

File: cargo/flash.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+esac
+
+export ESP_ARCH=
+{%- if mcu == "esp32c3" -%}
+riscv32imc-esp-espidf
+{%- else -%}
+xtensa-{{ mcu }}-espidf
+{%- endif %}
+
+web-flash --chip {{ mcu }} target/${ESP_ARCH}/${BUILD_MODE}/{{ crate_name }}
```
Comment (youduda) [inline cargo/flash.sh:27]: The Cargo.toml uses `name = "{{project-name}}"`. If a project name contains `-`, `{{crate_name}}` replaces that with a `_` and therefore flash.sh and run-wokwi.sh will not find the binary file.

**Comment by ivmarkov** [comment]
Thanks a lot!

**Comment by ivmarkov** [comment]
@SergioGasquez Would you mind looking at the CI failure and contributing a PR to fix it? Basically, you should also pass to the cargo-generate command lune build step all additional parameters you've introduced. Strange that this did not fail on the PR CI.

**Comment by SergioGasquez** [comment]
> @SergioGasquez Would you mind looking at the CI failure and contributing a PR to fix it? Basically, you should also pass to the cargo-generate command line build step all additional parameters you've introduced. Strange that this did not fail on the PR CI.

Ups! Weird that CI didn't catch it, just fixed it: #29

**Pull request merged**